### PR TITLE
extra code block escape hatch

### PR DESCRIPTION
**Pull request by Duhemm**
`Definition`s now accept an optional `append` configuration, whose value
will be appended at the end of the definition.

The parameter can be omitted. If present, it must be a single string or
an array of strings.

### CODE REVIEW

**Comment by eed3si9n** [comment]
What do you think about the name `extra` instead of `append`?

**Comment by Duhemm** [comment]
Much better 👍🏻

**Comment by eed3si9n** [comment]
Also it might make sense to add extra at the top, so you can introduce custom import statements in there.

**Comment by eed3si9n** [comment]
LGTM pending Travis

**Comment by Duhemm** [comment]
Just to make sure, did you mean adding another set of `extra` at the top of the file, or injecting code at the beginning of every definition?

**Comment by eed3si9n** [comment]
injecting code at the beginning of every definition.

File: library/src/main/scala/JavaCodeGen.scala [hunk 1]
```diff
@@ -72,6 +74,7 @@ class JavaCodeGen(lazyInterface: String, optionalInterface: String) extends Code
       s"""${genPackage(e)}
          |${genDoc(doc)}
          |public enum $name {
+         |    ${extra mkString EOL}
```
Comment (dwijnand) [inline library/src/main/scala/JavaCodeGen.scala]: I think this has to follow the values, and you need a semi-colon as well.. but this is just from memory, I could be miss-remembering.

Comment (eed3si9n) [inline library/src/main/scala/JavaCodeGen.scala]: You're right. I'm opening a continuation pull request.

**Comment by eed3si9n** [comment]
Removed LGTM comment.

**Comment by eed3si9n** [comment]
Closing this in favor of #31

**Pull request closed without merging**